core/domain/stats_jobs_test.py

```diff
@@ -341,3 +342,138 @@ def test_multiple_explorations(self):
                 'complete_exploration_count': 0,
                 'state_hit_counts': EMPTY_STATE_HIT_COUNTS_DICT,
             }, results)
+
+
+class ModifiedInteractionAnswerSummariesAggregator(
+        stats_jobs.StatisticsAggregator):
+    """A modified InteractionAnswerSummariesAggregator that does not start
+    a new batch job when the previous one has finished.
+    """
+    @classmethod
+    def _get_batch_job_manager_class(cls):
+        return ModifiedInteractionAnswerSummariesMRJobManager
+
+    @classmethod
+    def _kickoff_batch_job_after_previous_one_ends(cls):
+        pass
+
+
+class ModifiedInteractionAnswerSummariesMRJobManager(
+        stats_jobs.InteractionAnswerSummariesMRJobManager):
+
+    @classmethod
+    def _get_continuous_computation_class(cls):
+        return ModifiedInteractionAnswerSummariesAggregator
+
+
+class InteractionAnswerSummariesAggregatorTests(test_utils.GenericTestBase):
+    """Tests for interaction answer view aggregations."""
+
+    ALL_CONTINUOUS_COMPUTATION_MANAGERS_FOR_TESTS = [
+        ModifiedInteractionAnswerSummariesAggregator]
+    DEFAULT_RULESPEC_STR = exp_domain.DEFAULT_RULESPEC_STR
+
+    def _record_start(self, exp_id, exp_version, state_name, session_id):
+        event_services.StartExplorationEventHandler.record(
+            exp_id, exp_version, state_name, session_id, {},
+            feconf.PLAY_TYPE_NORMAL)
+
+    def test_one_answer(self):
+        with self.swap(
+                jobs_registry, 'ALL_CONTINUOUS_COMPUTATION_MANAGERS',
+                self.ALL_CONTINUOUS_COMPUTATION_MANAGERS_FOR_TESTS):
+
+            # setup example exploration
+            exp_id = 'eid'
+            exp = self.save_new_valid_exploration(exp_id, 'fake@user.com')
+            FIRST_STATE_NAME = exp.init_state_name
+            SECOND_STATE_NAME = 'State 2'
+            exp_services.update_exploration('fake@user.com', exp_id, [{
+                'cmd': 'edit_state_property',
```

Comment:
Ditto for 'cmd' and 'property_name' for these three updates (ie, please use exp_domain constants instead of strings).